src/inmanta/protocol/methods_v2.py

```diff
@@ -1804,6 +1804,45 @@ def discovered_resources_get_batch(
     """
 
 
+@auth(auth_label=const.CoreAuthorizationLabel.DISCOVERED_RESOURCES_DELETE, read_only=True, environment_param="tid")
+@typedmethod(
+    path="/discovered/<discovered_resource_id>",
+    operation="DELETE",
+    arg_options=methods.ENV_OPTS,
+    client_types=[ClientType.api],
+    api_version=2,
+)
+def discovered_resource_delete(
+    tid: uuid.UUID,
+    discovered_resource_id: str,
+) -> None:
+    """
+    Delete a discovered resource.
+
+    :param tid: The id of the environment this resource belongs to
+    :param discovered_resource_id: The id of the discovered_resource
+
+    :raise NotFound: This exception is raised when the referenced discovered resource is not found on the environment
```

Comment:
This message is used on the API documentation, where the name of this exception is translated to the corresponding return code 404. So the term "exception" doesn't apply there.
```suggestion
    :raise NotFound: When the referenced discovered resource is not found on the environment
```